### PR TITLE
Join implementations for BitSetLikes

### PR DESCRIPTION
**Pull request by Aceeri**

### CODE REVIEW

**Comment by torkleyy** [comment]
Isn't that what `CheckStorage` is (will be) for?

**Comment by Aceeri** [comment]
Sort of? This is kind of a superset of that where you can just use any `BitSetLike`, so you could unwrap the storage and then send in the bitset, which would make `CheckStorage` a convenience method really.

**Comment by torkleyy** [comment]
Okay, I think we can do that. But I would really want `check` to return a `BitSet` then.

**Comment by Aceeri** [comment]
I'll just include that change in this PR once I get a moment.

**Comment by Aceeri** [comment]
@torkleyy This should be good to go now.

File: src/bitset.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+use hibitset::{BitSet, BitSetAnd, BitSetLike, BitSetNot, BitSetOr};
+
+use ::{Index, Join};
+
+macro_rules! define_bit_join {
+    ( $bitset:ident [ $( $arg:ident ),* ] ) => {
+        impl<'a, $( $arg ),*> Join for &'a $bitset<$( $arg ),*>
```
Comment (torkleyy) [inline src/bitset.rs]: Do we also need `ParJoin` here? @WaDelma

Comment (WaDelma) [inline src/bitset.rs]: Yeah it looks like it.

**Comment by Aceeri** [comment]
Can we get this merged? I was hoping to have it merged before the next release.

**Pull request merged**